library/src/com/google/maps/android/kml/AbsKmlRenderer.java

```diff
@@ -99,11 +106,16 @@ private static boolean getPlacemarkVisibility(KmlPlacemark placemark) {
    * @param scale Scale value. A "1.0" scale value corresponds to the original size of the Bitmap
    * @return A scaled bitmap image
    */
-  private static BitmapDescriptor scaleIcon(Bitmap unscaledBitmap, Double scale) {
-    int width = (int) (unscaledBitmap.getWidth() * scale);
-    int height = (int) (unscaledBitmap.getHeight() * scale);
-    Bitmap scaledBitmap = Bitmap.createScaledBitmap(unscaledBitmap, width, height, false);
-    return BitmapDescriptorFactory.fromBitmap(scaledBitmap);
+  private static Bitmap scaleIcon(Context mContext, Bitmap unscaledBitmap, Double scale) {
+    int width = (int) convertDpToPixel(mContext, unscaledBitmap.getWidth() * scale);
+    int height = (int) convertDpToPixel(mContext, unscaledBitmap.getHeight() * scale);
+    return Bitmap.createScaledBitmap(unscaledBitmap, width, height, false);
+  }
+
+  private static double convertDpToPixel(Context context, double dp){
+    Resources resources = context.getResources();
+    DisplayMetrics metrics = resources.getDisplayMetrics();
+    return dp * ((float) metrics.densityDpi / DisplayMetrics.DENSITY_DEFAULT);
```

Comment:
`context.getResources().getDisplayMetrics().density * dp` for offsets, use rounding for sizes